docs/adr/0002-use-local-datastore.md

```diff
@@ -0,0 +1,58 @@
+# 2: Use local datastore for publisher workflow
+
+Date: 2018-07-20
+
+## Context
+We're building a new publishing architecture and MVP workflow in private beta. We have taken the decision to build this in a new application and therefore we need to decide how we use the publishing-api.
+
+One of the goals of the migration effort from 2016 was to for all publishing applications to be clients of the publishing-api and this API would means that these applications would not need local datastores. This is written up in the [GOV.UK Platform Architecture Goals 2017-18+][goals]
+
+[goals]: https://docs.google.com/document/d/1Oft4akc6dZfhhOjosNPbFpcLUOUjz7YG7QPcVZi8hww/edit#heading=h.5uytjxbfoe58
+
+Since writing these goals we have built the specialist-publisher application using this pattern. The difficulties of building Rails application this way have led us to reconsider this approach.
+
+We are about to build another publishing-api client and we should now use the learnings we have to ensure that not using a local datastore is the best decision.
+
+### Option 1: No local datastore
+In this option we continue with the platform architecture goals.
+
+#### Pros
++ Single source of truth - all data is in the publishing api
++ Multiple apps can edit the same content
++ Single implementation of scheduling and publishing logic
++ Single abstraction layer
+
+#### Cons
+- Effectively building an Active Resource layer
+- Harder to resolve performance problems
+- Current workflow is opinionated and may not match needs of content
+- Validation can be difficult to explain to user
+- Hard to deploy
+
+### Option 2: Use a local datastore
+#### Pros
++ It’s standard Rails so quick to build and easy to understand
++ Easy to test, testing over an API is not easy and slow
```

Comment:
None of the current publishing application test suites use the Publishing API, so I'm not sure how this applies? There is the Publishing End to end tests, but they explicitly are meant to use the Publishing API.

Reply:
This was in the context of the specialist publisher where the publishing api has to be mocked for a large part of the test suite.

What about changing the item to be:
> Easy to test, testing over a remote API involves a lot of stubbing or running a version of the API as part of the test suite.

Reply:
Updated in 4ce879a7b40632a614e2dff2c386aab6263dc8df